flink-connector-kudu/src/main/java/org/apache/flink/connectors/kudu/connector/reader/KuduReaderIterator.java

```diff
@@ -51,56 +52,63 @@ public boolean hasNext() throws KuduException {
         }
     }
 
-    public KuduRow next() {
+    public Row next() {
         RowResult row = this.rowIterator.next();
-        return toKuduRow(row);
+        return toFlinkRow(row);
     }
 
     private void nextRows() throws KuduException {
         this.rowIterator = scanner.nextRows();
     }
 
-    private KuduRow toKuduRow(RowResult row) {
+    private Row toFlinkRow(RowResult row) {
         Schema schema = row.getColumnProjection();
 
-        KuduRow values = new KuduRow(schema.getColumnCount());
+        Row values = new Row(schema.getColumnCount());
         schema.getColumns().forEach(column -> {
             String name = column.getName();
             int pos = schema.getColumnIndex(name);
-            if(row.isNull(name)) {
-                values.setField(pos, name, null);
+            if (row.isNull(name)) {
+                values.setField(pos, null);
             } else {
                 Type type = column.getType();
                 switch (type) {
                     case BINARY:
-                        values.setField(pos, name, row.getBinary(name));
+                        values.setField(pos, row.getBinary(name));
```

Comment:
Given `values.setField` takes an Object you can probably remove this switch and call `row.getObject(name)`.

Reply:
yes, lets simplify that!